### ISSUE TITLE
Deposit assets from trade page

### ISSUE THREAD

**Issue by jonator** [osmosis-labs/osmosis-frontend#655]
[Figma discussion](https://www.figma.com/file/wQjMyxY0EnEk29gBzGDMe5?node-id=4449:2029#216756478)

We should remove the roadblock for users loading the app for the first time having to go to the assets page to deposit before the go to the trade page to swap.

**Comment by jonator** [comment]
6/28 sprint planning -

Waiting on design work @victortokaito

**Comment by milesz8** [comment]
Is there a design you want implemented? I cant access the figma discussion. Otherwise I have some ideas

**Comment by jonator** [comment]
@milesz8 don't think we have a design for this yet. Or decided we want to do this.

We're currently undergoing a frontend redesign. May want to look into another issue for now.

**Comment by jonator** [comment]
One idea- what if when a user has a "from" asset selected with a positive balance, we can have the button change to deposit? They can click that, then the UI is locked until the tx completes, then the button says swap. Seems like a simple workflow that doesn't require another control @miles

**Comment by jonator** [comment]
9/29 meeting update -

We decided the best design would be to link to the assets page from the swap page if they have a 0 balance in Osmosis for the "from" asset. This would help inform users of the assets page, and would serve as a tutorial for how assets are deposited, withdrawn, and displayed.

It would work as follows from the swap page:
1. If a user has a 0 Osmosis balance for the asset in the "from" page, the swap button (or a link near it, design TBD) would show: "Deposit x ASSET from _network_" if they've approved the counterparty in Keplr, or "Deposit from _network_" otherwise
2. If they click the deposit button from the swap widget, we'd link to the assets page with query parameters in the URL to determine which transfer modal to have automatically opened.
3. User can carry out standard transfer from the transfer modal.
4. (Optional/stretch) when the transfer completes we'd keep the modal open and the "Transfer" button would be replaced with a "Back to Swap" button to let the user go back to the swap page with the query parameters populated to the relevant assets. (We've also considered having a link to Swap page in the success toast message)

**Comment by victortokaito** [comment]
['Begin a deposit from the swap page' Prototype](https://www.figma.com/file/om93lu8iXOhxwGHSUpAXEP/Trade-UI?node-id=667%3A5573)
* link to figma [frame](https://www.figma.com/file/om93lu8iXOhxwGHSUpAXEP/Trade-UI?node-id=667%3A6751)
